src/berry_pouch.c

```diff
@@ -0,0 +1,1528 @@
+#include "global.h"
+#include "malloc.h"
+#include "bg.h"
+#include "decompress.h"
+#include "gpu_regs.h"
+#include "palette.h"
+#include "text.h"
+#include "berry_pouch.h"
+#include "item_menu.h"
+#include "menu_helpers.h"
+#include "new_menu_helpers.h"
+#include "scanline_effect.h"
+#include "item_menu_icons.h"
+#include "list_menu.h"
+#include "graphics.h"
+#include "berry.h"
+#include "item.h"
+#include "item_use.h"
+#include "party_menu.h"
+#include "text_window.h"
+#include "strings.h"
+#include "string_util.h"
+#include "sound.h"
+#include "link.h"
+#include "money.h"
+#include "shop.h"
+#include "menu.h"
+#include "menu_indicators.h"
+#include "pokemon_storage_system.h"
+#include "constants/items.h"
+#include "constants/songs.h"
+
+struct BerryPouchStruct_203F36C
+{
+    void (*exitCallback)(void);
+    u16 indicatorOffset;
+    u8 indicatorTaskId;
+    u8 listMenuNumItems;
+    u8 listMenuMaxShowed;
+    u8 itemMenuIconId;
+    u8 ALIGNED(4) bg1TilemapBuffer[BG_SCREEN_SIZE];
+    s16 data[4];
+};
+
+struct BerryPouchStruct_203F370
+{
+    void (*savedCallback)(void);
+    u8 type;
+    u8 allowSelect;
+    u8 unused_06;
+    u16 listMenuSelectedRow;
+    u16 listMenuScrollOffset;
+};
+
+enum
+{
+    BP_ACTION_USE = 0,
+    BP_ACTION_TOSS,
+    BP_ACTION_GIVE,
+    BP_ACTION_EXIT,
+    BP_ACTION_DUMMY
+};
+
+static EWRAM_DATA struct BerryPouchStruct_203F36C *sResources = NULL;
+static EWRAM_DATA struct BerryPouchStruct_203F370 sStaticCnt = {};
+static EWRAM_DATA struct ListMenuItem *sListMenuItems = NULL;
+static EWRAM_DATA u8 *sListMenuStrbuf = NULL;
+static EWRAM_DATA const u8 *sContextMenuOptions = NULL;
+static EWRAM_DATA u8 sContextMenuNumOptions = 0;
+static ALIGNED(4) EWRAM_DATA u8 sVariableWindowIds[14] = {};
+static ALIGNED(4) EWRAM_DATA u8 sBerryPouchSpriteId = 0;
+
+static void CB2_InitBerryPouch(void);
+static bool8 RunBerryPouchInit(void);
+static void AbortBerryPouchLoading(void);
+static void Task_AbortBerryPouchLoading_WaitFade(u8 taskId);
+static void BerryPouchInitBgs(void);
+static bool8 BerryPouchLoadGfx(void);
+static bool8 AllocateListMenuBuffers(void);
+static void SetUpListMenuTemplate(void);
+static void GetBerryNameAndIndexForMenu(u8 * dest, u16 itemId);
+static void BerryPouchMoveCursorFunc(s32 itemIndex, bool8 onInit, struct ListMenu *list);
```

Comment:
Inconsistent space style for pointer. You usually add 1 space before and 1 after  `*`.

Reply:
Good catch. Fixed other instances to adhere to the following convention:
- No space between `*` and `const`
- No space between `struct StructName` or `union UnionName` and `*`
- Space between numeric type name and `*`